esmvalcore/cmor/_fixes/icon/icon.py

```diff
@@ -0,0 +1,163 @@
+"""On-the-fly CMORizer for ICON."""
+
+import logging
+from datetime import datetime
+
+import cf_units
+import dask.array as da
+import iris
+import numpy as np
+
+from esmvalcore.iris_helpers import var_name_constraint
+
+from ..fix import Fix
+from ..shared import add_scalar_height_coord, add_scalar_typesi_coord
+
+logger = logging.getLogger(__name__)
+
+
+class AllVars(Fix):
+    """Fixes for all variables."""
+
+    def fix_metadata(self, cubes):
+        """Fix metadata."""
+        raw_name = self.extra_facets.get('raw_name', self.vardef.short_name)
+        cube = cubes.extract_cube(var_name_constraint(raw_name))
+
+        # Fix dimensional coordinates
+        if cube.coords("time"):
+            self._fix_time(cube)
+        if cube.coords("height"):
+            plev_points_cube = cubes.extract_cube(var_name_constraint('pfull'))
+            plev_bounds_cube = cubes.extract_cube(var_name_constraint('phalf'))
+            cube = self._fix_height(cube, plev_points_cube, plev_bounds_cube)
+        lat_name = self.extra_facets.get('latitude', 'latitude')
+        lon_name = self.extra_facets.get('longitude', 'longitude')
+        if cube.coords(lat_name) and cube.coords(lon_name):
+            self._fix_lat_lon(cube, lat_name, lon_name)
```

Comment:
I remember that @bouweandela once told me (> 2 years ago) to use `self` instead of the class name :smile: Is there any particular reason to avoid `self`?

Reply:
:laughing: Well, for me it makes it clearer that this is not a method that can interact with self. But a bit of searching didn't give me any great reference. @bouweandela, do you have an opinion? I think if Bouwe doesn't way in, we shouldn't let this stop us.

Reply:
I still prefer a clearer distinction of static methods, but let's not consider it critical.